gui/src/main/java/org/jboss/as/console/mbui/widgets/ComplexAttributeForm.java

```diff
@@ -70,7 +77,17 @@ public ComplexAttributeForm(String attributeName, SecurityContext securityContex
 
     private ResourceDescription getAttributeDescription() {
         ModelNode desc = new ModelNode();
-        desc.get("attributes").set(resourceDescriptionDelegate.get("attributes").get(attributeName).get("value-type"));
+        desc.get(ATTRIBUTES).set(resourceDescriptionDelegate.get(ATTRIBUTES).get(attributeName).get(VALUE_TYPE));
+        boolean nillable = resourceDescriptionDelegate.get(ATTRIBUTES).get(attributeName).hasDefined(NILLABLE) ?
+                resourceDescriptionDelegate.get(ATTRIBUTES).get(attributeName).get(NILLABLE).asBoolean() : false;
+        if (nillable) {
```

Comment:
As described in the comment this is doing the opposite i.e. setting nillable=true (required=false) if the parent object is nillable. If it is not desirable to do it this way there still has to be a way to let the user set the whole object to null even if some of its nested attributes are required.